src/rest/server_service/error.rs

```diff
@@ -0,0 +1,43 @@
+use native_tls::Error as TlsError;
+use std::error::Error as StdError;
+use std::fmt::{Display, Formatter, Result as FmtResult};
+use std::io::Error as IoError;
+
+#[derive(Debug)]
+pub enum Error {
+    Pkcs12LoadFailed(IoError),
+    Pkcs12Invalid(TlsError),
+    BindFailed(IoError),
+    ServerAlreadyStopped,
+    ServerStopTimeout,
+    ServerStopFailed,
+}
+
+impl Display for Error {
+    fn fmt(&self, f: &mut Formatter) -> FmtResult {
+        let error_name = match *self {
```

Comment:
It's more conventional to use human-readable error messages in `Display`. Variant names are better suited for the `Debug` impl, which is macro-derived.

Reply:
Display must be implemented for Error, it's a direct replacement for `Error::description()`. I don't think it must be actually displayed and understandable by end-user, at least not at this point.

Reply:
Yes, but it should be human readable, no need to stuff things like enum variant names there.

Reply:
You're right, that's the point of a description. I'll fix it in next PR.

Reply:
This PR is not getting anywhere without being rebased to resolve the conflicts, so you might as well fix it here.

Reply:
This PR was already rebased and merged as a part of a different branch with additional functionalities. I would close or reuse this branch and this PR if I knew you were going to review it.

Reply:
OK, feel free to close this PR then.